v1/src/test/java/com/google/cloud/teleport/templates/TextToPubsubStreamLT.java

```diff
@@ -0,0 +1,174 @@
+import org.junit.runner.manipulation.Alphanumeric;
+import org.junit.runners.JUnit4;
+
+/** Performance tests for {@link TextToPubsubStreamLT GCS Text to PubSub} template. */
+@Category(TemplateLoadTest.class)
+@TemplateLoadTest(TextToPubsubStream.class)
+@RunWith(JUnit4.class)
+@OrderWith(Alphanumeric.class)
+public class TextToPubsubStreamLT extends TemplateLoadTestBase {
+  private static final String SPEC_PATH =
+      MoreObjects.firstNonNull(
+          TestProperties.specPath(),
+          "gs://dataflow-templates/latest/Stream_GCS_Text_to_Cloud_PubSub");
+  private static final String ARTIFACT_BUCKET = TestProperties.artifactBucket();
+  private static final String TEST_ROOT_DIR =
+      TextToPubsubStreamLT.class.getSimpleName().toLowerCase();
+  private static final String INPUT_PATH = "TextToPubsubStream/input.txt";
+  private static final String INPUT_PCOLLECTION =
+      "Read Text Data/Via ReadFiles/Read all via FileBasedSource/Read ranges.out0";
+
+  private static PubsubResourceManager pubsubResourceManager;
+  private static ArtifactClient gcsClient;
+  private static String generatorSchemaPath;
+
+  @Before
+  public void setup() throws IOException {
+    // Set up resource managers
+    pubsubResourceManager =
+        DefaultPubsubResourceManager.builder(testName, project)
+            .credentialsProvider(CREDENTIALS_PROVIDER)
+            .build();
+
+    Storage storageClient = createStorageClient(CREDENTIALS);
+
+    gcsClient = GcsArtifactClient.builder(storageClient, ARTIFACT_BUCKET, TEST_ROOT_DIR).build();
+    // upload schema files and save path
+    generatorSchemaPath =
+        getFullGcsPath(
+            ARTIFACT_BUCKET,
+            gcsClient
+                .uploadArtifact(
+                    "schema.json",
+                    Resources.getResource("TextToPubsubStream/schema.json").getPath())
+                .name());
+
+    // upload input.txt
+    gcsClient.uploadArtifact("input.txt", Resources.getResource(INPUT_PATH).getPath());
+  }
+
+  @After
+  public void teardown() {
+    ResourceManagerUtils.cleanResources(pubsubResourceManager, gcsClient);
+  }
+
+  @Test
+  public void test1Backlog10Mb() throws IOException {
+    testBacklog(Function.identity(), "520", 3, 20);
+  }
+
+  @Test
+  public void test2Backlog10Gb() throws IOException {
+    testBacklog(Function.identity(), "250000", 5, 50);
+  }
+
+  public void testBacklog(
+      Function<LaunchConfig.Builder, LaunchConfig.Builder> paramsAdder,
+      String qps,
+      long generatorTimeOut,
+      long pipelineTimeOut)
+      throws IOException {
+    // Creates a topic on Pub/Sub.
+    TopicName outputTopic = pubsubResourceManager.createTopic("topic");
+    // Creates a subscription on Pub/Sub.
+    SubscriptionName outputSubscription =
+        pubsubResourceManager.createSubscription(outputTopic, "output-subscription");
+
+    DataGenerator dataGenerator =
+        DataGenerator.builderWithSchemaLocation(testName, generatorSchemaPath)
```

Comment:
We can use the standard scheme `GAME_EVENT` here

Reply:
It's done.